packages/insomnia-app/app/plugins/context/app.tsx

```diff
@@ -69,7 +69,11 @@ export function init(renderPurpose: RenderPurpose = RENDER_PURPOSE_GENERAL): {
           return Promise.resolve(options.defaultValue || '');
         }
 
+        // This custom promise converts the prompt modal from being callback-based to reject when the modal is cancelled and resolve when the modal is submitted
         return new Promise((resolve, reject) => {
+          let shouldResolve = false;
+          let resolveWith: string;
```

Comment:
```suggestion
          let resolveWith: string | null = null;
```

this is going to need to be fixed when strict mode is on, best not to create more technical debt now.